### PR TITLE
Fix variable types modify in Entity init

### LINKED ISSUE #671: Entity.__init__ modifies `variable_types` dictionary
If an entity is being created using the optional `variable_types` parameter, the dictionary passed as `variable_types` will be modified to have the index column of the entity as a key.

Example Code:
```python
import featuretools as ft
import pandas as pd

df = pd.DataFrame({"id": [1, 2, 3, 4, 5, 6],
                   "card_id": [1, 2, 1, 3, 4, 5],
                   "transaction_time": [10, 12, 13, 20, 21, 20],
                   "fraud": [True, False, False, False, True, True]})

es = ft.EntitySet("fraud")

variable_types = {'fraud': ft.variable_types.Boolean}
print(variable_types)

es.entity_from_dataframe(entity_id="transactions",
                         dataframe=df,
                         index='id',
                         time_index='transaction_time',
                         variable_types=variable_types)

print(variable_types)
```

Output:
```
{'fraud': <class 'featuretools.variable_types.variable.Boolean'>}
{'fraud': <class 'featuretools.variable_types.variable.Boolean'>, 'id': <class 'featuretools.variable_types.variable.Index'>}
```

We should update the code in `Entity.__init__`to make a copy of the `variable_types` dictionary so that we do not changing the original dictionary

### PR DESCRIPTION
**Pull request by gsheni**
### Fix variable types modify in Entity init
Fixes #671

### CODE REVIEW

File: featuretools/tests/entityset_tests/test_entity.py [hunk 1]
```diff
@@ -127,3 +127,19 @@ def test_delete_variables(es):
     for var in to_delete:
         assert var not in variable_names
         assert var not in entity.df
+
+
+def test_variable_types_copy():
```
Comment (rwedge) [inline featuretools/tests/entityset_tests/test_entity.py]: Using copy is how we fixed the problem but I think the test name should describe the problem

File: docs/source/changelog.rst [hunk 2]
```diff
@@ -8,6 +8,7 @@ Changelog
     * Fixes
         * Fix performance regression in DFS (:pr:`637`)
         * Fix deserialization of feature relationship path (:pr:`665`)
+        * Fix variable types modify in Entity init (:pr:`675`)
```
Comment (rwedge) [inline docs/source/changelog.rst]: Can this message convey that we were fixing the user-supplied variable being modified?

Comment (gsheni) [reply docs/source/changelog.rst]: Fixed.

File: featuretools/tests/entityset_tests/test_entity.py (hunk 1 above)
Comment (gsheni) [reply featuretools/tests/entityset_tests/test_entity.py]: Fixed

**Pull request merged**